### PR TITLE
Fix: alarm 라우팅 오류

### LINKED ISSUE #116: fix: 알림 라우팅 오류수정
### 🐞 버그 설명

- 알림항목에 맞게 제대로 화면 이동이 안됨.

### 🛠️ 작업할 내용

- [] 관련 코드 수정

### 📷 스크린샷/로그

_No response_

### ✏️ 추가 내용

_No response_

### PR DESCRIPTION
**Pull request by taskcho123**
## 💡 Related issue

- closes #116 

## ✨ Description

- mock 데이터 사용하고 있던 부분 삭제
- 서버 link가 오는 경우 -> 라우팅 경로 / 아닌 경우 -> 연결하는 로직 분리
- 포그라운드 FCM 수신 시 unread count와 알림 목록도 재조회
- API 응답의 name, major, studentId, requestContent, tags를 전부 안전한 문자열/배열로 정규화

## 🔨 Implementation Lists
- [ ] API 연동  
       정규화 로직 추가해 연동 안정성 증가

## ✔️ Checklists

- [ ] 모든 테스트 통과
- [ ] 최신 develop 브랜치 merge 완료
- [ ] 다른 팀원 코드 수정 여부
- [ ] 문서 업데이트  
       ex. README에 “인증 플로우/라우트 가드 규칙” 및 에러코드 처리 규칙 추가

## 🔖 Uncompleted Tasks

- [ ] 후속 작업 예정  
       이벤트 관련 알림 로직 추후 추가 필요

## 📢 To Reviewers (필수)

1. 커피챗 요청, 팀원 요청, 팀원 승인요청 알림
=> 해당 유저 요청 받는 페이지로 넘어감
2. 커피챗 수락 =>  커피챗 페이지로 넘어감
4. 게시글 알림, 댓글 알림 => 해당 게시글 화면으로 이동
5. point 알림 => shop페이지로 이동

## Summary by CodeRabbit

* **New Features**
  * Added deep-link comment highlighting with smooth scrolling.
  * Improved notification navigation by resolving destinations with validation.
  * Added support for marking notifications as unread.
  * Introduced consistent press/tap interaction styling across key home cards/buttons.

* **Bug Fixes**
  * Chat UI now normalizes partner/text/tag data and treats “opponent exited” as unavailable; sending/typing is blocked when chat is unavailable.
  * Foreground notification updates now reliably refresh unread counts and lists (when signed in).
  * Notification click-to-read uses optimistic updates with rollback on failure.

* **Chores**
  * Centralized student label formatting and improved display fallbacks.

### CODE REVIEW

File: src/pages/home/notificationRouting.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  switch (notification.type) {
+    case 'coffeeChatRequest':
+    case 'teamApplicationReceived':
+      return notification.requestId
+        ? `/chat/requests/${notification.requestId}`
+        : '/chat/requests';
+    case 'coffeeChatAccepted':
+    case 'chatMessageReceived':
+      return '/chat';
+    case 'teamRecruitAccepted':
```
Comment (Mr-Planner) [inline src/pages/home/notificationRouting.ts:26]: case 'teamApplicationReceived'와 return 내용이 아예 같은데 case 'coffeChatRequest' 밑으로 옮겨도 될 것 같습니다 (가독성 문제)

**Review by Mr-Planner — CHANGES_REQUESTED** [review]
수고하셨습니다. 
리뷰 확인해주시고, 작업후에 다시 알려주세요 ! 

**1. PR 작성 내용과 다르게, 포인트 적립 알림 클릭 시 홈 화면으로 이동하고 있습니다.**

현재 코드상으로는 서버에서 내려오는 'link' 프로퍼티를 우선으로 하고 있는데 그 API 응답이 "/"로 되어있어서 홈 화면으로 이동하고 있습니다. 

BE분과 논의하셔서 수정하셔야겠으나... 개인적인 생각으로는 포인트 적립 알림 클릭시에도 다른 화면으로 이동하면 관련 알림 읽음처리하는 과정이 매우 불편했습니다. 
개인적으로는 이전처럼 클릭하고 알림에서만 사라지는 (화면 이동X) 방식이 좋아보입니다. 

**2. 디테일한 edge case들에 대한 동작이 개선되지 않았습니다.**

ex. 커피챗 종료 이후 커피챗 수락 요청을 읽었을때 / 내가 팔로잉 한 유저가 게시글 작성 후 삭제하고 나서 읽었을때 ...etc 

제가 노션 QA 문서에 시나리오 및 스크린샷을 첨부했습니다. 
확인해주시고, 이 밖의 다양한 알림 오류 시나리오들이 있을 것 같습니다. 
해당 노션 QA에 제가 언급한 2가지와 같은 타입의 edge case들을 적어주시고 Test 통과 했는지 여부를 적어주세요. 

그리고 이런 edge case들에 대한 알림 read 시 navigate는 전부 error type의 popup을 띄워서 navigate 못하게 막고 읽음처리 하는 것이 좋아보입니다.

File: src/pages/home/NotificationPage.tsx
Comment (Mr-Planner) [inline src/pages/home/NotificationPage.tsx:56]: Image

**Reponse API**
{"status":404,"code":48402,"message":"채팅방을 찾을 수 없습니다."}

-> 유저가 나간 채팅방에 대한 알림을 클릭했을때의 상황 추정

File: src/pages/home/NotificationPage.tsx [hunk 2]
```diff
@@ -61,6 +66,72 @@ const getErrorPopUpConfig = (status: number | null): PopUpConfig | null => {
   return null;
 };
 
+const getFallbackNavigationPopUpConfig = (): PopUpConfig => ({
```
Comment (Mr-Planner) [inline src/pages/home/NotificationPage.tsx]: **삭제된 게시글 클릭 시**

Image

**Review by Mr-Planner — COMMENTED** [review]
수고하셨습니다. 대부분 고쳐진 것 같네요,
이번 리뷰(3번)와 모두 읽음 기능 반영후에 승인하겠습니다.

1. 매 작업시 마다 구현 한 내용에 대해 주석 작성 부탁드려요

저같은 경우는 구현된 코드를 복습하면서 제가 이해한 내용을 주석으로 작성하고 있는데, 그렇게까진 아니어도 코드 설명용으로도 주석이 있어야 리뷰하는데 어려움이 없습니다. 

2. PR에 올려진 리뷰에 대한 답변은 디스코드 대신, 깃허브 comment로 작성해주시는게 나중에 협업관련 기록 관리하기에 좋을 것 같아요. 

3. Invalid한 경로의 라우팅은 거의 다 막힌 것 같은데, 팝업에 기록되는 내용에 대해 개선점이 있는 것 같습니다.

사진에 첨부한 것 처럼 특정 도메인 관련한 이동이 막힌 것인데, 전역적인 팝업문구가 보여지는 것이 유저에게는 딱딱하게 느껴질 것 같아요. 

개인적으로는 BE개발자분과 소통하여
- 서버가 취급하고 있는 에러사항을 다 파악
- 서버에서 보내는 status, code를 전부 파악하셔서
- status, code 조합에 맞는 팝업 문구를 전부 작성해두면
해결이 되지 않을까 싶습니다.

BE분께 응답으로 보내는 code가 특정 에러 상황에대한 고유한 code인지 확인후에 작업하시면 되지 않을까 싶습니다.

**Comment by taskcho123** [comment]
- 모두 읽음 기능 추가하였습니다
- 알림 에러 처리 분기 구체화
  커피챗/채팅방/댓글/게시글/모집글/포인트 알림별 이동 실패 문구 분리
  읽음 처리 실패도 403, 404, 500, 네트워크성 실패에 맞춰 문구 분리

**Review by Mr-Planner — APPROVED** [review]
수고하셨습니다 :)

1. 다만, 116브랜치와 133브랜치 모두 원격 develop브랜치에 반영된 내용을 merge하지 않고 작업 진행해주셨네요. 
항상 원격 develop에 반영된 내용들을 merge해주셔야 추후에 충돌문제가 생기지 않습니다. 이번 브랜치는 제가 병합했으니 추후 다른 PR은 merge 후에 올려주세요.
----

Image

2. 일부 테스트를 해보니 팝업문구가 너무 길어서 사진처럼 보기 좋지 않게 나오는 경우들이 있었습니다. 
구현 후에 충분한 테스트를 더 진행해주시면, 이런 UI 문제를 더 잘 잡을 수 있을거에요. 
문구는 제가 일괄적으로 수정해서 반영했습니다.
----

3. 커뮤니티의 MainHeader에서 뒤로 가기버튼이 /communitypage로 이동하도록 되어있습니다. 
그래서 커뮤니티 게시글 관련 알림으로 이동하고 화면 좌측 상단의 뒤로가기를 누르면 알림 화면 (사용자의 직전 행동)이 아닌, 커뮤니티 메인 페이지로 이동하는데 이는 직관적인 동작이 아니라고 생각합니다. 
**커뮤니티 화면에서의 헤더의 뒤로가기가 /communitypage로 이동하는 이유가 있을까요?**
별 다른 이유가 없다면 커뮤니티 이슈쪽에서 default 동작 (navigate(-1))으로 바꾸면 알림 클릭 경로로 진입후의 뒤로가기 결과가 자연스러울 것 같습니다.
